### PR TITLE
update to vs 2022 on building file

### PR DESCRIPTION
**Pull request by xHeaveny**
Pull request checklist:

- [x] `CHANGELOG.md` was updated, if applicable (don't think that's needed

# Description

### CODE REVIEW

File: BUILDING_ON_WINDOWS.md [hunk 1]
```diff
@@ -4,9 +4,9 @@
 
 This guide assumes you are on a 64-bit system. You might need to manually search out alternate download links should you desire to build Chatterino on a 32-bit system.
 
-## Visual Studio 2019
+## Visual Studio 2022
 
-Download and install [Visual Studio 2019 Community](https://visualstudio.microsoft.com/downloads/). In the installer, select "Desktop development with C++" and "Universal Windows Platform development".
+Download and install [Visual Studio 20122 Community](https://visualstudio.microsoft.com/downloads/). In the installer, select "Desktop development with C++" and "Universal Windows Platform development".
```
Comment (Mm2PL) [inline BUILDING_ON_WINDOWS.md]:
```suggestion
Download and install [Visual Studio 2012 Community](https://visualstudio.microsoft.com/downloads/). In the installer, select "Desktop development with C++" and "Universal Windows Platform development".
```
typo :)

**Comment by LosFarmosCTL** [comment]
There are some more reference to VS2019 in the instructions that should be updated as well if the main download is changed:

1. The VS-related boost dependencies. There should be information about what boost package to use for VS2022, plus updating the convenience link (outdated version anyways). I am unsure about which one is correct since I do not use visual studio myself, but from what I can tell on their website it should probably be msvc-14.3.
2. Under the instructions about what Qt components to install there is a reference to VS2019.
3. In "Run the build in Qt Creator" as well as "Producing standalone builds" there are a lot of references to the 2019 version inside of example paths

**Comment by xHeaveny** [comment]
> There are some more reference to VS2019 in the instructions that should be updated as well if the main download is changed:
> 
> 1. The VS-related boost dependencies. There should be information about what boost package to use for VS2022, plus updating the convenience link (outdated version anyways). I am unsure about which one is correct since I do not use visual studio myself, but from what I can tell on their website it should probably be msvc-14.3.
> 2. ~Under the instructions about what Qt components to install there is a reference to VS2019.~
> 3. ~In "Run the build in Qt Creator" as well as "Producing standalone builds" there are a lot of references to the 2019 version inside of example paths~
> 
> EDIT: While testing #3397 i found out that point 2 and 3 have not actually changed yet

I will change that than

**Comment by Mm2PL** [comment]
I find this pr a little suspicious

**Comment by xHeaveny** [comment]
> @xHeaveny why did you change the Qt version in the last commit, chatterino is not meant to (and wont even compile i think?) be built on Qt 6, only Qt 5 starting from version 12 or higher is supported right now.

it will compile but yea idk can change that

**Comment by xHeaveny** [comment]
wait but it works with Qt 6 and 5

**Comment by jammehcow** [comment]
Zneix put a lot of work in when he tried to migrate for Qt5 to Qt6 so I highly doubt it works. Please leave it at 5.

**Comment by xHeaveny** [comment]
> Zneix put a lot of work in when he tried to migrate for Qt5 to Qt6 so I highly doubt it works. Please leave it at 5.

it works 100% but yeah ofc I will change it to Qt5 again

**Comment by LosFarmosCTL** [comment]
> it works 100%

The project compiling successfully is not enough to call it a day and say that chatterino is now supporting Qt6. Before even thinking about updating build instructions, the entire CI pipeline would have to be moved to Qt6, preferably while dropping at least 5.12 support (not sure if backwards compatibility with 5.15 would be feasible), verify that the changes do not introduce any regressions on any platform (Qt6 got a new graphical backend) and best-case already switching out as much code as possible to newer Qt6 variations. 

Take a look at zneix work in #3287, it is definitely not something that can be introduced in a PR that updates the windows build instructions.

File: BUILDING_ON_WINDOWS.md [hunk 2]
```diff
@@ -84,7 +84,7 @@ Compiling with Breakpad support enables crash reports that can be of use for dev
 
 1. Open the `chatterino.pro` file by double-clicking it, or by opening it via Qt Creator.
 2. You will be presented with a screen that is titled "Configure Project". In this screen, you should have at least one option present ready to be configured, like this:
-   ![Qt Create Configure Project screenshot](https://user-images.githubusercontent.com/41973452/159462759-470e5371-671e-478e-85ca-33452ca9bea3.png)
+   ![Qt Create Configure Project screenshot](https://i.imgur.com/dbz45mB.png)
```
Comment (pajlada) [inline BUILDING_ON_WINDOWS.md:87]: 1. The image should use githubusercontent (just paste it into a github comment and you can get a link
2. The image downgrades to VS 2017 and Qt 5.11, it should use one of our supported versions of Qt5 and VS 2022

**Review by pajlada — CHANGES_REQUESTED** [review]

File: BUILDING_ON_WINDOWS.md [hunk 3]
```diff
@@ -20,14 +20,14 @@ Notes:
    - Visit the downloads list on [SourceForge](https://sourceforge.net/projects/boost/files/boost-binaries/).
    - Select the latest version from the list.
    - Download the `.exe` file appropriate to your Visual Studio installation version and system bitness (choose `-64` for 64-bit systems).
-     Visual Studio versions map as follows: `14.2` in the filename corresponds to MSVC 2019, `14.1` to 2017, `14.0` to 2015. _Anything prior to Visual Studio 2015 is unsupported. Please upgrade should you have an older installation._
+     Visual Studio versions map as follows: `14.3` in the filename corresponds to MSVC 2022,`14.2`to 2019, `14.1` to 2017, `14.0` to 2015. _Anything prior to Visual Studio 2015 is unsupported. Please upgrade should you have an older installation._
```
Comment (zneix) [inline BUILDING_ON_WINDOWS.md:23]:
```suggestion
     Visual Studio versions map as follows: `14.3` in the filename corresponds to MSVC 2022,`14.2` to 2019, `14.1` to 2017, `14.0` to 2015. _Anything prior to Visual Studio 2015 is unsupported. Please upgrade should you have an older installation._
```

File: BUILDING_ON_WINDOWS.md (hunk 2 above)
Comment (xHeaveny) [reply BUILDING_ON_WINDOWS.md:87]: i can do it when i am back at home yes

**Review by pajlada — CHANGES_REQUESTED** [review]
Very much nitpicking at this point, but could you take a screenshot of your Qt Creator as a smaller window?

See 
image
(top is yours, bottom is old) for reference. Old one was actually readable, while yours gets scaled down and becomes unreadable unless you click into the image

**Review by zneix — APPROVED** [review]

**Comment by xHeaveny** [comment]
now it shoud be readable

**Review by pajlada — APPROVED** [review]

**Pull request merged**